### PR TITLE
correct missing self parameters, remove duplicate class methods

### PR DESCRIPTION
**Pull request by Josverl**
Hi,
I stumbled across your work and would like to thank you for the stub you created.

I found a few thinks that i think should be corrected, 
which were revealed by suppressing mostly irrelevant pylint errors

I'm working on a set of stubs from micropython , and although i do not have a pyboard myself, support for the pyb should be included.

Is it OK with you to re-use your work for that ?

regards,
Jos

### CODE REVIEW

File: lib/pyb.py [hunk 1]
```diff
@@ -982,13 +981,6 @@ def write(self, buf):
         """
         pass
 
-    def writechar(char):
```
Comment (dastultz) [inline lib/pyb.py:985]: According to this document, this method is still good:
https://docs.micropython.org/en/latest/library/pyb.UART.html

File: lib/pyb.py [hunk 2]
```diff
@@ -200,9 +202,6 @@ def __init__(self, pin):
         """
         pass
 
-    def read_timed(self):
-        pass
-
     def read_timed_stop(self):
```
Comment (dastultz) [inline lib/pyb.py:205]: I don't see this method in this doc any more: https://docs.micropython.org/en/latest/library/pyb.ADC.html

**Comment by dastultz** [comment]
Mostly everything looks good, the linter and self work is fine. Can you make another pass through the current docs and make sure we are deleting stuff that should be and not deleting stuff that shouldn't?

Thanks for the help!

**Comment by Josverl** [comment]
Hi Daryl,
Tried to review my own work , and found one edge case where it is really not clear `Pin.af_list() ` and which I cannot test myself lacking the hardware.

I think the rest checks out pretty well. Please see dfacd55 for documentation / rationale

**Review by dastultz — CHANGES_REQUESTED** [review]
Hi Jos, I'm still waiting for you to address my two comments, you removed a method that looks like it is still supported, and didn't remove on that looks like it is no longer supported.

File: lib/pyb.py (hunk 2 above)
Comment (Josverl) [reply lib/pyb.py:205]: i am not a pyboard user, and do not follow updates/removals.
same as the above , it was definede twice ( incorrect and correct) 
i just removed the mistake as i think was clearly documented.

I you want me to review all prior code and double check them against the online documentation for a board I do not even own, then please understand that this is not my priority, and therefore i will retract this pull request.

Regards,
Jos

Comment (dastultz) [reply lib/pyb.py:205]: I understand that you removed "read_timed" because it was defined a second time. This comment is about "read_timed_stop". I was asking why you didn't remove "read_timed_stop". I thought you might have good reason to keep it in and I was looking for that.
